pypit/artrace.py

```diff
@@ -1359,265 +1365,543 @@ def model_tilt(slf, det, msarc, censpec=None, maskval=-999999.9,
-            if guesstilts is not None:
-                factr += guesstilts[arcdet[j], ordcen[arcdet[j], 0]]*float(msarc.shape[0]-1.0)
-            idx = int(factr+0.5)
+            factr = (msarc.shape[0] - 1.0) * arutils.func_val(mcoeff, ordcen[arcdet[j], 0],
+                                                              settings.argflag['trace']['slits']['function'],
+                                                              minv=0.0, maxv=msarc.shape[1] - 1.0)
+            idx = int(factr + 0.5)
             if (idx > 0) and (idx < msarc.shape[0]):
                 maskrows[idx] = 0
                 tcoeff[:, idx] = mcoeff.copy()
-                weights[idx] = (np.abs(np.median(apfit[wmask])))**0.25
             # Restrict to good IDs?
             if settings.argflag['trace']['slits']['tilts']['idsonly']:
                 if not aduse[j]:
                     maskrows[idx] = 1
 
-            # if True:
-            #     #debugger.set_trace()
-            #     plt.clf()
-            #     plt.plot(xtfit[wmask], ytfit[wmask], 'bx')
-            #     plt.plot(xtfit, model, 'r-')
-            #     plt.show()
-            #     continue
-
-            xint = int(xtfit[0])
-            xtilt[xint:xint+2*sz+1, j] = xtfit/(msarc.shape[1]-1.0)
-            ytilt[xint:xint+2*sz+1, j] = arcdet[j]/(msarc.shape[0]-1.0)
-            ztilt[xint:xint+2*sz+1, j] = ytfit/(msarc.shape[0]-1.0)
-            stilt[xint:xint+2*sz+1, j] = shfit
+            xtilt[xint:xint + 2 * sz + 1, j] = xtfit / (msarc.shape[1] - 1.0)
+            ytilt[xint:xint + 2 * sz + 1, j] = arcdet[j] / (msarc.shape[0] - 1.0)
+            ztilt[xint:xint + 2 * sz + 1, j] = ytfit / (msarc.shape[0] - 1.0)
             if settings.argflag['trace']['slits']['tilts']['method'].lower() == "spline":
-                mtilt[xint:xint+2*sz+1, j] = model/(msarc.shape[0]-1.0)
+                mtilt[xint:xint + 2 * sz + 1, j] = model / (msarc.shape[0] - 1.0)
             elif settings.argflag['trace']['slits']['tilts']['method'].lower() == "interp":
-                mtilt[xint:xint+2*sz+1, j] = (2.0*model[sz]-model)/(msarc.shape[0]-1.0)
+                mtilt[xint:xint + 2 * sz + 1, j] = (2.0 * model[sz] - model) / (msarc.shape[0] - 1.0)
             else:
-                mtilt[xint:xint+2*sz+1, j] = (2.0*model[sz]-model)/(msarc.shape[0]-1.0)
+                mtilt[xint:xint + 2 * sz + 1, j] = (2.0 * model[sz] - model) / (msarc.shape[0] - 1.0)
             wbad = np.where(ytfit == maskval)[0]
-            ztilt[xint+wbad, j] = maskval
+            ztilt[xint + wbad, j] = maskval
             if wmask.size != 0:
-                sigg = max(1.4826 * np.median(np.abs(ytfit-model)[wmask])/np.sqrt(2.0), 1.0)
-                wtilt[xint:xint+2*sz+1, j] = 1.0 / sigg
+                sigg = max(1.4826 * np.median(np.abs(ytfit - model)[wmask]) / np.sqrt(2.0), 1.0)
+                wtilt[xint:xint + 2 * sz + 1, j] = 1.0 / sigg
             # Extrapolate off the slit to the edges of the chip
```

Comment:
This weighting only appears to be used by the `spline` function for tracing the tilts (I think the default is either `spca` or `interp`), so I don't think anyone uses `spline` as the standard.

I don't remember the reason for why the weight is inverse sigma (instead of inverse variance), but the idea is that more weight is given to better traced (usually brighter) arc lines. In other words, a large sigg value corresponds to data that has a large scatter about the model tilt (and it therefore is weighted lower).